### DISCUSSION TITLE
Richiesta di chiarimenti sull'Attestazione di Conformità.

### DISCUSSION THREAD

**Discussion by gcuci** [italia/ansc#787] [Supporto servizi cooperativi] [answered]
Con il rilascio della versione '1.26.0' dei servizi, e della versione '100010' dell'interfaccia, si è introdotta l'obbligatorietà di effettuare, da parte dell'USC, la firma dell'Attestazione di Conformità, prima della firma dell'evento.

Non ci è chiarissimo, se questo nuovo step di firma, è sempre obbligatorio in presenza di allegati all'evento, o se ci sono situazioni in cui può non essere previsto. Nella 'Guida_operativa_ANSC', descrivendone il dettaglio, nel contesto della Web Application, se ne indica la necessità, in presenza di documenti cartacei successivamente digitalizzati tramite scansione. Ma con l'attuale servizio di upload degli allegati non sembra si possa passare questa informazione. Inoltre, il servizio di Anteprima Attestazione, restituisce, oltre al pdf dell'anteprima, anche la proprietà 'hasAllegatiDaAttestare', che fa pensare ad un'eventuale differenziazione in base al tipo di allegati. 

La domanda è: possiamo considerare obbligatorio lo step di firma dell'Attestazione di Conformità da parte dell'USC, in presenza di allegati all'evento, o è comunque necessario verificare il valore della proprietà 'hasAllegatiDaAttestare', restituita dal servizio di Anteprima Attestazione, che potrebbe eventualmente dare un'indicazione differente?

In attesa di riscontro,

cordiali saluti.

**Comment by siscomanpr** [comment]
Ci accodiamo anche noi alla richiesta di chiarimenti di @gcuci.
Non ci è chiaro se il nuovo indicatore flagAttestazione di ModelAllegatoRif è un campo che dobbiamo inviarvi noi sulla base della scelta che fa l'utente a video, oppure se è un dato gestito da voi. E se lo gestite voi, è possibile sapere in base a cosa capite/decidete che un allegato è da attestare oppure no?

Grazie
Mauro Maccagno
Siscom SPA

**Comment by mttfranci** [comment, marked as answer]
@gcuci  L'attuale comportamento, leggermente modificato con l'ultimo rilascio e che dovrebbe essere quello definitivo è il seguente : 

L'attestazione di conformità va firmata solo se un qualsiasi documento è stato allegato all'evento (compresa eventualmente la firma dichiarante analogica, id tipo allegato = 99). Questo vale per qualsiasi allegato, anche qualora sia già stato firmato digitalmente.
In tali casi il servizio /anteprima/attestazione/{version} restituierà l'attributo hasAllegatiDaAttestare = true ed eventualmente genererà l'anteprima.
 
Se invece non è presente nessun allegato, il servizio /anteprima/attestazione/{version} restituierà l'attributo hasAllegatiDaAttestare = false e non genererà in nessun caso l'anteprima.
 
Non è necessario invocare il servizio di anteprima per sapere se l'attestazione di conformità (hasAllegatiDaAttestare = true) sia necessaria oppure no. Se l'evento è formato interamente da gestionale e già sapete che non ci sono allegati potete saltare l'anteprima e la firma dell'attestazione di conformità e andare direttamente alla firma dell'USC.
 
@siscomanpr potete ignorare il campo flagAttestazione dello schema ModelAllegatoRif.

Cercheremo di integrare anche la documentazione.

Siamo a disposizione per ulteriori chiarimenti.

Saluti
